src/provider/Provider.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { createContext, useState } from 'react';
+
+const MyContext = createContext<any>(null);
+
+type ContextValueType = {
+  addedIDS: number[];
+  cartValue: number;
+  handleClick: (id: number) => void;
+};
+
+type MyContextProviderProps = {
+  children: React.ReactNode;
+};
+
+export const MyContextProvider: React.FC<MyContextProviderProps> = ({
+  children,
+}) => {
+  const [addedIDS, setAddedIDS] = useState<number[]>([]);
+  const [cartValue, setCartValue] = useState<number>(0);
+
+  const handleClick = (id: number) => {
```

Comment:
i think it will be good to change name of handleClick to somthing like add or handle addToCart to better represent its function